sim/adversary/adversary.go

```diff
@@ -10,10 +10,12 @@ type Receiver interface {
 // Endpoint with which the adversary can control the network
 type Host interface {
 	gpbft.Host
+	// Sends a message to all other participants
+	Broadcast(sender gpbft.ActorID, msg *gpbft.GMessage)
```

Comment:
I think this method is unnecessary. All callers use the message sender as the sender parameter. Now that this PR is delivering messages locally, I don't think we need a sender parameter at all, not in BroadcastSynchronous either.